src/gui/mainwindow.h

```diff
@@ -52,61 +58,47 @@ class downloadFromURL;
 class LineEdit;
 class ExecutionLog;
 class PowerManagement;
-
-QT_BEGIN_NAMESPACE
-class QCloseEvent;
-class QFileSystemWatcher;
-class QShortcut;
-class QSplitter;
-class QTabWidget;
-class QTimer;
-QT_END_NAMESPACE
+class StatsDialog;
 
 namespace BitTorrent
 {
     class TorrentHandle;
 }
 
-class MainWindow: public QMainWindow, private Ui::MainWindow
+namespace Ui
+{
+    class MainWindow;
+}
+
+class MainWindow : public QMainWindow
 {
     Q_OBJECT
 
 public:
-    // Construct / Destruct
     explicit MainWindow(QWidget *parent = 0);
-    ~MainWindow();
-    // Methods
-    QWidget* getCurrentTabWidget() const;
-    TransferListWidget* getTransferList() const { return transferList; }
-    QMenu* getTrayIconMenu();
-    PropertiesWidget *getProperties() const { return properties; }
-
-public slots:
-    void trackerAuthenticationRequired(BitTorrent::TorrentHandle *const torrent);
-    void setTabText(int index, QString text) const;
-    void showNotificationBaloon(QString title, QString msg) const;
-    void downloadFromURLList(const QStringList& urls);
-    void updateAltSpeedsBtn(bool alternative);
-    void updateNbTorrents();
+    ~MainWindow() override;
+
+    QWidget* currentTabWidget() const;
+    TransferListWidget* transferListWidget() const;
+    PropertiesWidget *propertiesWidget() const;
+    QMenu* trayIconMenu();
+
     void activate();
     void cleanup();
 
-protected slots:
-    // GUI related slots
+    void showNotificationBaloon(QString title, QString msg) const;
```

Comment:
Aha! I was mislead by the diff layout, where it is shown under protected section. I apologise.